sdk/src/cameras/itof-camera/camera_itof.cpp

```diff
@@ -1288,121 +1267,27 @@ static int16_t getValueFromJSON(cJSON *config_json, std::string key) {
-                          << m_ccb_calibrationFile;
-            } else {
-                LOG(WARNING) << "Duplicate calibration file ignored: "
-                             << json_ccb_calibration_file->valuestring;
-            }
-        }
+    if (m_initConfigFilePath == "") {
 
-        // Get depthparameter list location
-        const cJSON *json_depth_ini_file = nullptr;
-        json_depth_ini_file =
-            cJSON_GetObjectItemCaseSensitive(config_json, "depthIni");
-        if (cJSON_IsString(json_depth_ini_file) == false) { // Check for old key
-            json_depth_ini_file =
-                cJSON_GetObjectItemCaseSensitive(config_json, "DEPTH_INI");
-        }
-        if (cJSON_IsString(json_depth_ini_file) &&
-            (json_depth_ini_file->valuestring != NULL)) {
-            // store depthparameter list location
-            std::string mode;
-            std::vector<std::string> iniFiles;
-
-            if (m_ini_depth.empty()) {
-                Utils::splitIntoTokens(
-                    std::string(json_depth_ini_file->valuestring), ';',
-                    iniFiles);
-                if (iniFiles.size() > 1) {
-                    for (const std::string &file : iniFiles) {
-                        //extract last string that is after last underscore (e.g. 'mp' will be extracted from ini_file_mp)
-                        size_t lastUnderscorePos = file.find_last_of("_");
-                        if (lastUnderscorePos == std::string::npos) {
-                            LOG(WARNING) << "File: " << file
-                                         << " has no suffix that can be used "
-                                            "to identify the mode";
-                            continue;
-                        }
-
-                        size_t dotPos = file.find_last_of(".");
-                        mode = file.substr(lastUnderscorePos + 1,
-                                           dotPos - lastUnderscorePos - 1);
-                        // TO DO: check is mode is supported by the camera
-
-                        LOG(INFO) << "Found Depth parameter list: " << file;
-                        // Create map with mode name as key and path as value
-                        m_ini_depth_map.emplace(mode, file);
-                    }
-                    saveDepthParamsToJsonFile(
-                        "./config/user_config.json"); // TODO: for testing only, remove before 5.1.0 release
-                    // Set m_ini_depth to first map element
-                    auto it = m_ini_depth_map.begin();
-                    m_ini_depth = it->second;
-                } else {
-                    m_ini_depth = std::string(json_depth_ini_file->valuestring);
-                }
-
-                LOG(INFO) << "Current Depthparameter list is: " << m_ini_depth;
-            }
-        }
+        for (auto it = m_availableModes.begin(); it != m_availableModes.end();
```

Comment:
you can also use: 
for (const auto &mode : m_availableModes) {
Then you no longer need this line:
"int mode = *it;